openmls/src/extensions/mod.rs

```diff
@@ -195,6 +195,183 @@ pub enum Extension {
     Lifetime(LifetimeExtension),
 }
 
+/// A list of extensions with unique extension types.
+#[derive(
+    Debug, Clone, PartialEq, Eq, Serialize, Deserialize, TlsSize, TlsSerialize, TlsDeserialize,
+)]
+pub struct Extensions {
+    inner: Vec<Extension>,
+}
+
+impl Extensions {
+    /// Create an extension list that is empty.
+    pub fn empty() -> Self {
+        Self { inner: Vec::new() }
+    }
+
+    /// Create an extension list that contains a single extension.
+    pub fn single(extension: Extension) -> Self {
+        Self {
+            inner: vec![extension],
+        }
+    }
+
+    /// Create an extension list that contains multiple extensions.
+    ///
+    /// This function will fail when the list of extensions contains duplicate extension types.
+    pub fn multi(extensions: Vec<Extension>) -> Result<Self, &'static str> {
+        extensions.try_into()
+    }
+
+    // ---------------------------------------------------------------------------------------------
+
+    /// Add an extension to the extension list.
+    ///
+    /// Returns an error when there already is an extension with the same extension type.
+    pub fn add(&mut self, extension: Extension) -> Result<(), &'static str> {
+        if !self.contains(extension.extension_type()) {
+            self.inner.push(extension);
+            Ok(())
+        } else {
+            Err("Cannot add duplicate extension.")
+        }
+    }
+
+    /// Add an extension to the extension list (or silently replace an existing one.)
+    pub fn add_or_replace(&mut self, extension: Extension) {
+        let _ = self.remove(extension.extension_type());
+        self.add(extension).unwrap();
+    }
+
+    /// Remove an extension from the extension list.
+    ///
+    /// Returns an error when there is no extension with the given extension type.
+    pub fn remove(&mut self, extension_type: ExtensionType) -> Result<(), &'static str> {
```

Comment:
Remove functions are generally expected to return the value they removed. I don't think we need it anywhere right now. But we should probably keep the convention.